gcloud-java-logging/src/main/java/com/google/gcloud/logging/spi/v2/ConfigServiceV2Api.java

```diff
@@ -0,0 +1,533 @@
+   * same name that is specified in `sinkName`.  If the sink does not
+   * exist, it is created.
+   * @param sink The updated sink, whose name must be the same as the sink
+   * identifier in `sinkName`.  If `sinkName` does not exist, then
+   * this method creates a new sink.
+   */
+  public LogSink updateSink(String sinkName, LogSink sink) {
+    UpdateSinkRequest request =
+        UpdateSinkRequest.newBuilder()
+        .setSinkName(sinkName)
+        .setSink(sink)
+        .build();
+
+    return updateSink(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates or updates a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   */
+  public LogSink updateSink(UpdateSinkRequest request) {
+    return updateSinkCallable().call(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Creates or updates a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public ApiCallable<UpdateSinkRequest, LogSink> updateSinkCallable() {
+    return ApiUtils.prepareIdempotentCallable(UPDATE_SINK, settings).bind(channel);
+  }
+
+  // ----- deleteSink -----
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param sinkName The resource name of the sink to delete.
+   * Example: `"projects/my-project-id/sinks/my-sink-id"`.
+   */
+  public void deleteSink(String sinkName) {
+    DeleteSinkRequest request =
+        DeleteSinkRequest.newBuilder()
+        .setSinkName(sinkName)
+        .build();
+
+    deleteSink(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   *
+   * @param request The request object containing all of the parameters for the API call.
+   */
+  public void deleteSink(DeleteSinkRequest request) {
+    deleteSinkCallable().call(request);
+  }
+
+  // AUTO-GENERATED DOCUMENTATION AND METHOD - see instructions at the top of the file for editing.
+  /**
+   * Deletes a sink.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  public ApiCallable<DeleteSinkRequest, Empty> deleteSinkCallable() {
+    return ApiUtils.prepareIdempotentCallable(DELETE_SINK, settings).bind(channel);
+  }
+
+
+  // ========
+  // Cleanup
+  // ========
+
+  /**
+   * Initiates an orderly shutdown in which preexisting calls continue but new calls are immediately
+   * cancelled.
+   *
+   * <!-- manual edit -->
+   * <!-- end manual edit -->
+   */
+  @Override
+  public void close() {
+    // Manually-added shutdown code
```

Comment:
I am not sure if this is worth it. The manual sections are there so that if the people who generate the API need a more sophisticated shutdown sequence, they can add it. Delegating this to another method, in my humble opinion, just adds more methods with no real gain.